### DISCUSSION TITLE
Verona : The form component needs to have a UIForm in its ancestry.

### DISCUSSION THREAD

**Discussion by ritonglue** [primefaces/community#2557] [PrimeFaces Templates] [answered]
If I open:
http://www.primefaces.org/verona/crud.xhtml

I'll get this message:

`Warning
The form component needs to have a UIForm in its ancestry. Suggestion: enclose the necessary components within <h:form>`

**Comment by melloware** [comment]
That sounds like the CRUD page is missing a closing `</h:form>` tag.  I don't own the Verona template but from looking at my templates I do have my crud.xhtml is correct and has a closing tag?  So probably a bug in that Verona that is deployed but should be easy for you to fix locally.

**Reply by melloware** [reply]
If you send me a DM on the Discord channel I can help you update Verona to PF14 and fix that error if interested.

**Reply by ritonglue** [reply]
I don't own the Verona template either. I was just browsing the show page and the error popped out.
I hope somebody will fix it in the future.

**Reply by melloware** [reply, marked as answer]
If anyone else reads this the bug is in `topbar.xhtml` the menu input needs to be wrapped in a form.

```xml
<h:form>
       <p:inputText type="text" placeholder="Search" />
       <span class="pi pi-search"></span>
</h:form>
```

**Comment by tandraschko** [comment]
fixed for the upcoming 15.0.1
